nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/events/MoleculeDataFormatHandler.java

```diff
@@ -0,0 +1,88 @@
+package org.nmrfx.analyst.gui.events;
+
+import org.nmrfx.analyst.gui.molecule.CanvasMolecule;
+import org.nmrfx.analyst.gui.molecule.MoleculeUtils;
+import org.nmrfx.chemistry.io.MoleculeIOException;
+import org.nmrfx.chemistry.io.SDFile;
+import org.nmrfx.processor.gui.CanvasAnnotation;
+import org.nmrfx.processor.gui.FXMLController;
+import org.nmrfx.processor.gui.PolyChart;
+import org.nmrfx.processor.gui.events.DataFormatEventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+/**
+ * Class to handle clipboard MDLCT DataFormat.
+ */
+public class MoleculeDataFormatHandler implements DataFormatEventHandler {
+    private static final Logger log = LoggerFactory.getLogger(MoleculeDataFormatHandler.class);
+
+    /**
+     * Attempt to add a molecule in MDCLT format to the canvas.
+     * @param o ByteBuffer containing molecule information in MDCLT format.
+     */
+    @Override
+    public void handlePaste(Object o) {
+
+        // Don't try to paste the molecule if there is no dataset displayed
+        PolyChart activeChart = FXMLController.getActiveController().getActiveChart();
+        if (activeChart.getDataset() == null) {
+            return;
+        }
+        String molContents = parseMdlctBuffer((ByteBuffer)  o);
+        // Not a mol file without multiple lines
+        if (molContents.indexOf('\n') < 0) {
+            return;
+        }
+        // Use the first line of the string as the filename
+        String filename = molContents.split("\n")[0];
```

Comment:
this may result in empty string filenames -> blank molecule names, not sure how to name the molecule if the title string isn't set otherwise..

Reply:
discussed with @brucejohnson and added a prompt for a molecule name if there was no text on the title line of the pasted molecule file